vulnerable_people_form/tier_lookup/lacode_tier_loookup.py

```diff
@@ -0,0 +1,66 @@
+import bios
+import subprocess
+from shutil import copyfile, rmtree
+from _datetime import datetime
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+LOCAL_RESTRICTIONS_FILE = "local-restrictions.yaml"
+PATH_TO_RESTRICTIONS_FILE = f"vulnerable_people_form/tier_lookup/{LOCAL_RESTRICTIONS_FILE}"
+COLLECTIONS_GIT_REPO = "https://github.com/alphagov/collections.git"
+DEFAULT_NATIONAL_ALERT_LEVEL = 3
+
+
+class LacodeTierLookup:
+    """
+    fetches the tier status for supplied LA Code
+    """
+
+    def __init__(self,
+                 load_tier_data: bool = False,
+                 tier_file_path: str = None):
+
+        self.tier_file_path = tier_file_path
+        if load_tier_data:
+            self.fetch_tier_data_file()
+
+    def fetch_tier_data_file(self):
+        args = ["git", "clone", "--depth=1", COLLECTIONS_GIT_REPO]
+        logger.info(f"Cloning Git repo - {COLLECTIONS_GIT_REPO}")
+        res = subprocess.Popen(args, stdout=subprocess.PIPE)
+        output, _error = res.communicate()
+        logger.info(f"Finished Cloning Git repo - {COLLECTIONS_GIT_REPO}")
+
+        if not _error:
+            copyfile(f"collections/lib/local_restrictions/{LOCAL_RESTRICTIONS_FILE}", PATH_TO_RESTRICTIONS_FILE)
+            rmtree("collections")
+        else:
+            logger.error(f"Unable to fetch new local restrictions file, defaulting to existing file: {_error}")
+
+    def lacode_to_tier(self, lacode: str, date: datetime = datetime.now()):
+        if self.tier_file_path is None:
+            self.tier_file_path = LOCAL_RESTRICTIONS_FILE
+
+        my_dict = bios.read(self.tier_file_path)
+        all_las = my_dict[1]
+        try:
+            lacode_record = all_las[lacode]
+        except KeyError:
+            logger.error(f"LA code {lacode} not present in restrictions file. Returning default")
+            return DEFAULT_NATIONAL_ALERT_LEVEL
+
+        sorted_restrictions_by_date = sorted(lacode_record["restrictions"], key=lambda k: k['start_date'], reverse=True)
+
+        for restriction in sorted_restrictions_by_date:
+            if date > self.date_time_from_restriction_dict(restriction=restriction):
+                return restriction["alert_level"]
+
+        logger.error(f"LA code {lacode} has no relevant restriction level. Returning default")
+        return DEFAULT_NATIONAL_ALERT_LEVEL
+
+    def date_time_from_restriction_dict(self, restriction: dict):
```

Comment:
Can this method be called `restriction_comes_in_to_force_datetime` or something?

Reply:
Fixed